### PR TITLE
feat(Core/Conf): add CONFIG_ARENA_QUEUE_ANNOUNCER_DETAIL

### PR DESCRIPTION
**Pull request by r-o-b-o-t-o**
## Changes Proposed:
- Add `Arena.QueueAnnouncer.Detail` to the world conf
```properties
#    Arena.QueueAnnouncer.Detail
#        Description: The amount of detail on teams queued for arenas to announce.
#        Default:     3 - (Announce the team's name and rating)
#                     2 - (Announce only the team's name)
#                     1 - (Announce only the team's rating)
#                     0 - (Do not announce any information about the teams)
#

Arena.QueueAnnouncer.Detail = 0
```
- Added french locale for the arena queue announcer

## Issues Addressed:

N/A

## SOURCE:

N/A

## Tests Performed:

- Built and tested on Windows 11

## How to Test the Changes:

1. Change the value of `Arena.QueueAnnouncer.Detail` in the world config
2. Queue for rated arena matches
3. Look at the announcement in chat

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Review by Gultask — APPROVED** [review]

**Review by Kitzunu — COMMENTED** [review]
You missing the implementation of CONFIG_ARENA_QUEUE_ANNOUNCER_DETAIL part :P

**Comment by r-o-b-o-t-o** [comment]
> You missing the implementation of CONFIG_ARENA_QUEUE_ANNOUNCER_DETAIL part :P

https://github.com/azerothcore/azerothcore-wotlk/pull/16850/files#diff-d3adf9f4592b73d74171ddbc2b7fa3cd8a1a2868bdb0d18bf060a6806f69be07R1146-R1161

https://github.com/azerothcore/azerothcore-wotlk/pull/16850/files#diff-d3adf9f4592b73d74171ddbc2b7fa3cd8a1a2868bdb0d18bf060a6806f69be07R1186-R1201

**Comment by Kitzunu** [comment]
Oh what the fuck, I could only see 1 cpp file at first O.O

File: src/server/game/Miscellaneous/Language.h [hunk 1]
```diff
@@ -666,8 +666,14 @@ enum AcoreStrings
...
-    LANG_ARENA_QUEUE_ANNOUNCE_WORLD_JOIN = 718,
-    LANG_ARENA_QUEUE_ANNOUNCE_WORLD_EXIT = 719,
+    LANG_ARENA_QUEUE_ANNOUNCE_WORLD_JOIN_NAME_RATING = 718,
+    LANG_ARENA_QUEUE_ANNOUNCE_WORLD_EXIT_NAME_RATING = 719,
+    LANG_ARENA_QUEUE_ANNOUNCE_WORLD_JOIN_NAME = 753,
+    LANG_ARENA_QUEUE_ANNOUNCE_WORLD_EXIT_NAME = 754,
+    LANG_ARENA_QUEUE_ANNOUNCE_WORLD_JOIN_RATING = 755,
+    LANG_ARENA_QUEUE_ANNOUNCE_WORLD_EXIT_RATING = 756,
+    LANG_ARENA_QUEUE_ANNOUNCE_WORLD_JOIN = 757,
+    LANG_ARENA_QUEUE_ANNOUNCE_WORLD_EXIT = 758,
```
Comment (Kitzunu) [inline src/server/game/Miscellaneous/Language.h:676]: Ideally, these should be placed in numeric order for the whole file to not mistakenly do double entries or remove stuff.

**Review by Kitzunu — APPROVED** [review]

**Review by Nyeriah — APPROVED** [review]

**Comment by r-o-b-o-t-o** [comment]
> Sql still faili it seems

Don't know what's up with it

**Comment by Nyeriah** [comment]
Hm.. yeah, I am at loss too

**Comment by michaeldelago** [comment]
@r-o-b-o-t-o

It's failing because a bug in the script. 

https://github.com/azerothcore/azerothcore-wotlk/blob/9eadbdf6119b25a6953f93b0e357ae607837f21b/apps/ci/ci-pending.sh#L17

That `sed` is seeing the "--" in the middle of the text and thinks its a SQL comment. then discards everything after it. 

```console
$ sed 's/ --[^--]*$//' data/sql/updates/pending_db_world/reaching.sql | tail -n2
UPDATE `acore_string` SET `locale_frFR` = '|cffff0000[Annonce File d''Attente Arène]:|r %s
UPDATE `acore_string` SET `locale_frFR` = '|cffff0000[Annonce File d''Attente Arène]:|r %s                                            
$ tail -n2 data/sql/updates/pending_db_world/reaching.sql
UPDATE `acore_string` SET `locale_frFR` = '|cffff0000[Annonce File d''Attente Arène]:|r %s -- Rejoint : %ux%u : %u|r' WHERE `entry` = 718;
UPDATE `acore_string` SET `locale_frFR` = '|cffff0000[Annonce File d''Attente Arène]:|r %s -- Quitté : %ux%u : %u|r' WHERE `entry` = 719;
```

**Pull request merged**